plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,508 @@
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Slurm struct {
+	URL              string          `toml:"url"`
+	Username         string          `toml:"username"`
+	Token            string          `toml:"token"`
+	IgnoredEndpoints []string        `toml:"ignored_endpoints"`
+	ResponseTimeout  config.Duration `toml:"response_timeout"`
+	tls.ClientConfig
+
+	client      *goslurm.APIClient
+	baseURL     *url.URL
+	endpointMap map[string]bool
+}
+
+func (*Slurm) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Slurm) Init() error {
+	if s.ResponseTimeout == 0 {
+		s.ResponseTimeout = config.Duration(time.Second * 5)
+	}
+
+	s.endpointMap = map[string]bool{}
+	for _, endpoint := range s.IgnoredEndpoints {
+		s.endpointMap[strings.ToLower(endpoint)] = true
+	}
+
+	if s.URL == "" {
+		return errors.New("empty URL provided")
+	}
+
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		return err
+	}
+
+	if u.Hostname() == "" {
+		return fmt.Errorf("empty hostname for url %q", s.URL)
+	}
+
+	s.baseURL = u
+
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return fmt.Errorf("invalid scheme %q", u.Scheme)
+	}
+
+	var tlsCfg *stdTls.Config
+	if u.Scheme == "https" {
+		tlsCfg, err = s.ClientConfig.TLSConfig()
+		if err != nil {
+			return err
+		}
+	}
+
+	configuration := goslurm.NewConfiguration()
+	configuration.Host = u.Host
+	configuration.Scheme = u.Scheme
+	configuration.UserAgent = internal.ProductToken()
+	configuration.HTTPClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+		},
+		Timeout: time.Duration(s.ResponseTimeout),
+	}
+
+	s.client = goslurm.NewAPIClient(configuration)
+
+	return nil
+}
+
+func (s *Slurm) parseTres(tres string) map[string]interface{} {
+	parsedValues := map[string]interface{}{}
+	var ok bool
+
+	for _, tresVal := range strings.Split(tres, ",") {
+		parsedTresVal := strings.Split(tresVal, "=")
+		if len(parsedTresVal) != 2 {
+			continue
+		}
+
+		tag := parsedTresVal[0]
+		val := parsedTresVal[1]
+		var factor float64 = 1
+
+		if tag == "mem" {
+			factor, ok = map[string]float64{
+				"K": 1.0 / 1024.0,
+				"M": 1,
+				"G": 1024,
+				"T": 1024 * 1024,
+			}[strings.ToUpper(val[len(val)-1:])]
+			if !ok {
+				continue
+			}
+			val = val[:len(val)-1]
+		}
+
+		parsedInt, err := strconv.ParseInt(val, 10, 64)
+		if err == nil && factor >= 1 {
+			parsedValues[tag] = parsedInt * int64(factor)
+			continue
+		}
+		parsedFloat, err := strconv.ParseFloat(val, 64)
+		if err == nil {
+			parsedValues[tag] = parsedFloat * factor
+			continue
+		}
+		parsedValues[tag] = val
+	}
+
+	return parsedValues
+}
+
+func (s *Slurm) gatherDiagMetrics(acc telegraf.Accumulator,
+	diag *goslurm.V0038DiagStatistics) {
```

Comment:
```suggestion
func (s *Slurm) gatherDiagMetrics(acc telegraf.Accumulator, diag *goslurm.V0038DiagStatistics) {
```

Reply:
If I try to apply the suggestion manually I get a syntax error:

    missing ',' before newline in parameter list

What's more, if I add a trailing comma so that the function declaration reads

    func (s *Slurm) gatherDiagMetrics(acc telegraf.Accumulator, diag,
    *goslurm.V0038DiagStatistics) {

I get yet another syntax error:

    mixed named and unnamed parameters

Finally, if I move the `diag` argument to col 0 like so

    func (s *Slurm) gatherDiagMetrics(acc telegraf.Accumulator,
    diag *goslurm.V0038DiagStatistics) {

Then `gofmt` will tabulate the line so that it ends up just like in the original...

Would you rather have me 'ignore' `gofmt` and leave `diag` at col 0?

Sorry for all the hassle...

Reply:
You can click the apply suggestion button. Otherwise please see that there is no newline in the suggestion of Sven, he wants all to be on 1 line.

Reply:
Whoops! My bad... thanks a ton for pointing that out! I didn't spot the missing `+` on the second line :(

I broke that up into two lines to try to stick to the 80 chars-per-line limit.

Should the same be applied to the other `Gather*` function definitions?

Thanks again for the heads up!

Reply:
AFAICS there is no hard limit on *.go files..

Reply:
And yeah, I thin the intention is indeed to also apply this to the other `gather*` methods..

Reply:
For the sake of completeness, I found this 80 character suggestion [here](https://github.com/influxdata/telegraf/blob/master/docs/developers/CODE_STYLE.md?plain=1#L7-L8) 😜

I'll apply the fix to the other functions as well: thanks a ton for confirming the intention 😄

Reply:
Anyway, these have all been fixed on 6096a8891396cd38c18a853afa9a626031ccbe78. Thanks again!